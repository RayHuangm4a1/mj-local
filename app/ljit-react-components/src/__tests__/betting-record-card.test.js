import React from 'react';
import { shallow, mount, } from 'enzyme';
import BettingRecordCard from '../components/betting-record-card';

describe('Betting Record Card', () => {
	it('handle default props', () => {
		const {
			onClick,
			onClickCancel,
			bettingRecord,
			className,
		} = BettingRecordCard.defaultProps;

		expect(onClick).toBeDefined();
		expect(onClick).toBeInstanceOf(Function);
		expect(onClickCancel).toBeDefined();
		expect(onClickCancel).toBeInstanceOf(Function);
		expect(bettingRecord).toEqual({});
		expect(className).toBe('');
	});

	it('should renders correctly', () => {
		const wrapper = shallow(
			<BettingRecordCard
				bettingRecord={{
					id: 201901240027,
					name: 'mock-name',
					lotteryId: 0,
					lotteryName: 'mock-lottery',
					playId: 1,
					amountPerBet: 10,
					multiple: 2,
					reward: 0,
					rebate: 12.8,
					issue: '20190701-0001',
					opencode: '',
					count: 1,
					betcontent: '和',
					weizhi: '',
					details: [ // 中獎資訊
						{
							name: '和',
							count: 1,
							reward: 10,
						},
					],
					status: BettingRecordCard.StatusEnums.NOT_OPENED,
					createdAt: new Date('2019-04-13T10:38:43'),
				}}
				className='mock-class'
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-betting-info-card', () => {
		const wrapper = shallow(<BettingRecordCard/>);

		expect(wrapper.hasClass('ljit-betting-info-card')).toEqual(true);
	});

	it('should contains class ljit-betting-info-card__info', () => {
		const wrapper = shallow(<BettingRecordCard/>);

		expect(wrapper.find('.ljit-betting-info-card__info')).toHaveLength(1);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<BettingRecordCard className={className}/>);

		expect(wrapper.hasClass(className)).toEqual(true);
	});
	it('should handle onClick event', () => {
		const onClick = jest.fn();
		const wrapper = mount(
			<BettingRecordCard
				onClick={onClick}
			/>
		);

		wrapper.simulate('click');
		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalled();
	});
	it('should handle onClickCancel event', () => {
		const onClickCancel = jest.fn();
		const wrapper = mount(
			<BettingRecordCard
				onClickCancel={onClickCancel}
				bettingRecord={{ status: BettingRecordCard.StatusEnums.NOT_OPENED, }}
			/>
		);

		wrapper.find('button').simulate('click');
		expect(onClickCancel).toHaveBeenCalledTimes(1);
		expect(onClickCancel).toHaveBeenCalled();
	});

	it('should mount in a full DOM', () => {
		const bettingRecord = {
			id: 201901240027,
			name: 'mock-name',
			lotteryId: 0,
			lotteryName: 'mock-lottery',
			playId: 1,
			amountPerBet: 10,
			multiple: 2,
			reward: 0,
			rebate: 12.8,
			issue: '20190701-0001',
			opencode: '',
			count: 1,
			betcontent: '和',
			weizhi: '',
			details: [ // 中獎資訊
				{
					name: '和',
					count: 1,
					reward: 10,
				},
			],
			status: BettingRecordCard.StatusEnums.NEW,
			createdAt: new Date('2019-04-13T10:38:43'),
		};
		const className = 'mock-class';
		const onClick = () => {};
		const onClickCancel = () => {};
		const wrapper = mount(
			<BettingRecordCard
				bettingRecord={bettingRecord}
				className={className}
				onClick={onClick}
				onClickCancel={onClickCancel}
			/>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().bettingRecord).toMatchObject(bettingRecord);
		expect(wrapper.props().onClick).toEqual(onClick);
		expect(wrapper.props().onClickCancel).toEqual(onClickCancel);
	});
});
