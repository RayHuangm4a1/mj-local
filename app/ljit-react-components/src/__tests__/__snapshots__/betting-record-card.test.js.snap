// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Betting Record Card should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BettingRecordCard
    bettingRecord={
      Object {
        "amountPerBet": 10,
        "betcontent": "和",
        "count": 1,
        "createdAt": 2019-04-13T02:38:43.000Z,
        "details": Array [
          Object {
            "count": 1,
            "name": "和",
            "reward": 10,
          },
        ],
        "id": 201901240027,
        "issue": "20190701-0001",
        "lotteryId": 0,
        "lotteryName": "mock-lottery",
        "multiple": 2,
        "name": "mock-name",
        "opencode": "",
        "playId": 1,
        "rebate": 12.8,
        "reward": 0,
        "status": 8,
        "weizhi": "",
      }
    }
    className="mock-class"
    onClick={[Function]}
    onClickCancel={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <InfoCard
        className="ljit-betting-info-card__info"
        isShowingCloseButton={false}
        left={
          <React.Fragment>
            <LabelText
              fontSize="small"
              label="方案号"
              labelColType="medium"
              labelStyle={Object {}}
              text={201901240027}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="玩法"
              labelColType="medium"
              labelStyle={Object {}}
              text="mock-name"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="单注金额"
              labelColType="large"
              labelStyle={Object {}}
              text={10}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="中奖金额"
              labelColType="large"
              labelStyle={Object {}}
              text="-"
              textStyle={Object {}}
            />
          </React.Fragment>
        }
        onClose={[Function]}
        right={
          <React.Fragment>
            <LabelText
              fontSize="small"
              label="期号"
              labelColType="small"
              labelStyle={Object {}}
              text="20190701-0001"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="时间"
              labelColType="small"
              labelStyle={Object {}}
              text="4/13 10:38:43"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="倍数"
              labelColType="small"
              labelStyle={Object {}}
              text={2}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="中奖号码"
              labelColType="large"
              labelStyle={Object {}}
              text="-"
              textStyle={Object {}}
            />
          </React.Fragment>
        }
        topLeft={
          <div
            className="ljit-betting-info-card__header"
          >
            <span
              className="ljit-betting-info-card__title"
            >
              mock-lottery
            </span>
            <TextButton
              className="ljit-betting-info-card__cancel"
              color="default"
              disabled={false}
              fontSize="small"
              fontWeight="heavy"
              onClick={[Function]}
              text="撤單"
            />
            <StatusTag
              backgroundColor=""
              status="betUnOpened"
              tagType="default"
              text="未开奖"
              textColor=""
            />
          </div>
        }
      />,
      "className": "ljit-betting-info-card mock-class",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "ljit-betting-info-card__info",
        "isShowingCloseButton": false,
        "left": <React.Fragment>
          <LabelText
            fontSize="small"
            label="方案号"
            labelColType="medium"
            labelStyle={Object {}}
            text={201901240027}
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="玩法"
            labelColType="medium"
            labelStyle={Object {}}
            text="mock-name"
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="单注金额"
            labelColType="large"
            labelStyle={Object {}}
            text={10}
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="中奖金额"
            labelColType="large"
            labelStyle={Object {}}
            text="-"
            textStyle={Object {}}
          />
        </React.Fragment>,
        "onClose": [Function],
        "right": <React.Fragment>
          <LabelText
            fontSize="small"
            label="期号"
            labelColType="small"
            labelStyle={Object {}}
            text="20190701-0001"
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="时间"
            labelColType="small"
            labelStyle={Object {}}
            text="4/13 10:38:43"
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="倍数"
            labelColType="small"
            labelStyle={Object {}}
            text={2}
            textStyle={Object {}}
          />
          <LabelText
            fontSize="small"
            label="中奖号码"
            labelColType="large"
            labelStyle={Object {}}
            text="-"
            textStyle={Object {}}
          />
        </React.Fragment>,
        "topLeft": <div
          className="ljit-betting-info-card__header"
        >
          <span
            className="ljit-betting-info-card__title"
          >
            mock-lottery
          </span>
          <TextButton
            className="ljit-betting-info-card__cancel"
            color="default"
            disabled={false}
            fontSize="small"
            fontWeight="heavy"
            onClick={[Function]}
            text="撤單"
          />
          <StatusTag
            backgroundColor=""
            status="betUnOpened"
            tagType="default"
            text="未开奖"
            textColor=""
          />
        </div>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <InfoCard
          className="ljit-betting-info-card__info"
          isShowingCloseButton={false}
          left={
            <React.Fragment>
              <LabelText
                fontSize="small"
                label="方案号"
                labelColType="medium"
                labelStyle={Object {}}
                text={201901240027}
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="玩法"
                labelColType="medium"
                labelStyle={Object {}}
                text="mock-name"
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="单注金额"
                labelColType="large"
                labelStyle={Object {}}
                text={10}
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="中奖金额"
                labelColType="large"
                labelStyle={Object {}}
                text="-"
                textStyle={Object {}}
              />
            </React.Fragment>
          }
          onClose={[Function]}
          right={
            <React.Fragment>
              <LabelText
                fontSize="small"
                label="期号"
                labelColType="small"
                labelStyle={Object {}}
                text="20190701-0001"
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="时间"
                labelColType="small"
                labelStyle={Object {}}
                text="4/13 10:38:43"
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="倍数"
                labelColType="small"
                labelStyle={Object {}}
                text={2}
                textStyle={Object {}}
              />
              <LabelText
                fontSize="small"
                label="中奖号码"
                labelColType="large"
                labelStyle={Object {}}
                text="-"
                textStyle={Object {}}
              />
            </React.Fragment>
          }
          topLeft={
            <div
              className="ljit-betting-info-card__header"
            >
              <span
                className="ljit-betting-info-card__title"
              >
                mock-lottery
              </span>
              <TextButton
                className="ljit-betting-info-card__cancel"
                color="default"
                disabled={false}
                fontSize="small"
                fontWeight="heavy"
                onClick={[Function]}
                text="撤單"
              />
              <StatusTag
                backgroundColor=""
                status="betUnOpened"
                tagType="default"
                text="未开奖"
                textColor=""
              />
            </div>
          }
        />,
        "className": "ljit-betting-info-card mock-class",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "ljit-betting-info-card__info",
          "isShowingCloseButton": false,
          "left": <React.Fragment>
            <LabelText
              fontSize="small"
              label="方案号"
              labelColType="medium"
              labelStyle={Object {}}
              text={201901240027}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="玩法"
              labelColType="medium"
              labelStyle={Object {}}
              text="mock-name"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="单注金额"
              labelColType="large"
              labelStyle={Object {}}
              text={10}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="中奖金额"
              labelColType="large"
              labelStyle={Object {}}
              text="-"
              textStyle={Object {}}
            />
          </React.Fragment>,
          "onClose": [Function],
          "right": <React.Fragment>
            <LabelText
              fontSize="small"
              label="期号"
              labelColType="small"
              labelStyle={Object {}}
              text="20190701-0001"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="时间"
              labelColType="small"
              labelStyle={Object {}}
              text="4/13 10:38:43"
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="倍数"
              labelColType="small"
              labelStyle={Object {}}
              text={2}
              textStyle={Object {}}
            />
            <LabelText
              fontSize="small"
              label="中奖号码"
              labelColType="large"
              labelStyle={Object {}}
              text="-"
              textStyle={Object {}}
            />
          </React.Fragment>,
          "topLeft": <div
            className="ljit-betting-info-card__header"
          >
            <span
              className="ljit-betting-info-card__title"
            >
              mock-lottery
            </span>
            <TextButton
              className="ljit-betting-info-card__cancel"
              color="default"
              disabled={false}
              fontSize="small"
              fontWeight="heavy"
              onClick={[Function]}
              text="撤單"
            />
            <StatusTag
              backgroundColor=""
              status="betUnOpened"
              tagType="default"
              text="未开奖"
              textColor=""
            />
          </div>,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
